### ISSUE TITLE
Delay undefined task execution

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda/zeebe#12993]
**Describe the bug**

Related to #12957 

Straight-through processing loops are a problem. These mostly occur when there is a loop of undefined tasks. In #12957 we will come up with a proper fix for this issue. This specific issues targets a quick-fix to resolve the most common problem.

We should delay the execution of an undefined task in a process. This will prevent the loop from overburdening the Broker. For this there are 2 options:

1. We delay it synchronously (aka `Thread.sleep`)
2. We delay it asynchronously

Synchronously is the quickest and dirtiest solution. A downside is that this could impact the overall processing latency/throughput.

Asynchronous is prettier, but a lot more effort. We'd have to split the activation and completion of undefined tasks in two pieces. the initial processing will take care of activation. We'll have to introduce a scheduler which will check for uncompleted undefined tasks and send complete commands for these.

**To Reproduce**

Create a process with a loop of undefined tasks.

**Expected behavior**

The loop doesn't overload the cluster

**Comment by megglos** [comment]
A sync block is not an option, as such a process will degrade the performance of a whole partition. This is not easily transparent to the user and will require manual effort to root-cause, thus only the async solution should be considered.

With this constraint and in context of https://github.com/camunda/zeebe/issues/12957
How does this mitigation compare in terms of complexity to a validation that detects loops without a single wait state element?

**Comment by remcowesterhoud** [comment]
Closing this, as we chose a different band-aid solution

**Comment by oleschoenburg** [comment]
@remcowesterhoud Hm, but async delay wouldn't have been a band aid I think. Are we sure that we will stick with loop detection and will not implement async delay?

**Comment by remcowesterhoud** [comment]
@oleschoenburg no, but the main issue is still open and async delay is listed as one of the option there and contains more details. Hence why I closed this one.
